python/cudf/cudf/core/column/column.py

```diff
@@ -1911,20 +1911,17 @@ def as_column(
         if dtype is not None:
             column = column.astype(dtype)
         return column
-    elif isinstance(arbitrary, ColumnBase):
-        if dtype is not None:
-            return arbitrary.astype(dtype)
+    elif isinstance(arbitrary, (ColumnBase, cudf.Series, cudf.BaseIndex)):
+        if isinstance(arbitrary, cudf.Series):
+            column = arbitrary._column
+        elif isinstance(arbitrary, cudf.BaseIndex):
+            column = arbitrary._values
         else:
-            return arbitrary
-    elif isinstance(arbitrary, cudf.Series):
-        data = arbitrary._column
-        if dtype is not None:
-            data = data.astype(dtype)
-    elif isinstance(arbitrary, cudf.BaseIndex):
-        data = arbitrary._values
+            column = arbitrary
+        # TODO: Should nan_as_null apply here?
```

Comment:
Good question. I think ignoring it is probably right since that flag is primarily designed to help with ingestion from sources that don't properly support differentiating NaN and null (aka pandas, especially at the time this code was written). I'll defer to @shwina in case he has different thoughts though.

Reply:
Yeah I think that's right. Although it may then be helpful to add something like this in the docstring:

```
nan_as_null: bool
    Use N/A in the output where there are NaNs in the input. This parameter only applies when arbitrary is a <...>. 
```

Reply:
Thanks for confirming. Updated the docstring.